src/esp/physics/bullet/BulletPhysicsManager.cpp

```diff
@@ -642,5 +641,275 @@ std::vector<ContactPointData> BulletPhysicsManager::getContactPoints() const {
+                                     std::move(fixedConstraint));
+    }
+  }
+
+  // link objects to their consraints for later deactivation/removal logic
+  objectConstraints_[settings.objectIdA].push_back(nextConstraintId_);
+  if (settings.objectIdB != ID_UNDEFINED) {
+    objectConstraints_[settings.objectIdB].push_back(nextConstraintId_);
+  }
+
+  // use the updater to set params
+  updateRigidConstraint(nextConstraintId_, settings);
+
+  return nextConstraintId_++;
+}
+
+void BulletPhysicsManager::updateRigidConstraint(
+    int constraintId,
+    const RigidConstraintSettings& settings) {
+  // validate that object and link ids are unchanged for update.
+  ESP_CHECK(rigidConstraintSettings_.count(constraintId) > 0,
+            "::updateRigidConstraint - Provided invalid constraintId = "
+                << constraintId);
+  auto& cachedSettings = rigidConstraintSettings_.at(constraintId);
+  ESP_CHECK(cachedSettings->objectIdA == settings.objectIdA,
+            "::updateRigidConstraint - RigidConstraintSettings::objectIdA must "
+            "match existing settings ("
+                << settings.objectIdA << " vs. " << cachedSettings->objectIdA
+                << ")");
+  ESP_CHECK(cachedSettings->objectIdB == settings.objectIdB,
+            "::updateRigidConstraint - RigidConstraintSettings::objectIdB must "
+            "match existing settings ("
+                << settings.objectIdB << " vs. " << cachedSettings->objectIdB
+                << ")");
+  ESP_CHECK(cachedSettings->linkIdA == settings.linkIdA,
+            "::updateRigidConstraint - RigidConstraintSettings::linkIdA must "
+            "match existing settings ("
+                << settings.linkIdA << " vs. " << cachedSettings->linkIdA
+                << ")");
+  ESP_CHECK(cachedSettings->linkIdB == settings.linkIdB,
+            "::updateRigidConstraint - RigidConstraintSettings::linkIdB must "
+            "match existing settings ("
+                << settings.linkIdB << " vs. " << cachedSettings->linkIdB
+                << ")");
+  ESP_CHECK(cachedSettings->constraintType == settings.constraintType,
+            "::updateRigidConstraint - RigidConstraintSettings::constraintType "
+            "must match existing settings ("
+                << int(settings.constraintType) << " vs. "
+                << int(cachedSettings->constraintType) << ")");
+
+  if (articulatedP2PConstraints_.count(constraintId) > 0) {
+    // NOTE: oddly, pivotA cannot be set through the API for this constraint
+    // type.
+    ESP_CHECK(cachedSettings->pivotA == settings.pivotA,
+              "::updateRigidConstraint - RigidConstraintSettings::pivotA must "
+              "match existing settings for multibody P2P constraints. Instead, "
+              "remove and create to update this parameter. ("
+                  << settings.pivotA << " vs. " << cachedSettings->pivotA
+                  << ")");
+    // TODO: Either fix the Bullet API or do the add/remove for the user here.
```

Comment:
No, this still applies. Oddly you cannot set the `btMultiBodyPoint2Point::pivotInA`. It is protected with no setter. I don't see any reason for this, just an oversight I think. However, I don't want to modify our Bullet submodule at this moment. the alternative is to quietly remove and re-create the constraint if user provides a new pivot. Thought it would be better to make this extra cost obvious for now instead.